packages/file/src/upload.coffee.md

```diff
@@ -0,0 +1,155 @@
+          description: """
+          File path where to write content to. Pass the content.
+          """
+      required: ['source', 'target']
+
+## Handler
+
+    handler = ({config, log}) ->
+      if config.md5?
+        algo = 'md5'
+      else if config.sha1?
+        algo = 'sha1'
+      else
+        algo = 'md5'
+      log message: "Source is \"#{config.source}\"", level: 'DEBUG', module: 'nikita/lib/file/upload'
+      log message: "Destination is \"#{config.target}\"", level: 'DEBUG', module: 'nikita/lib/file/upload'
+      log message: "Entering file.upload", level: 'DEBUG', module: 'nikita/lib/file/upload'
+      # Stat the target and redefine its path if a directory
+      stats = await @call raw_output: true, ->
+        try
+          {stats} = await @fs.base.stat ssh: false, sudo: false, target: config.target
+          # Target is a file
+          return stats if utils.stats.isFile stats.mode
+          # Target is invalid
+          throw Error "Invalid Target: expect a file, a symlink or a directory for #{JSON.stringify config.target}" unless utils.stats.isDirectory stats.mode
+          # Target is a directory
+          config.target = path.resolve config.target, path.basename config.source
+          try
+            {stats} = await @fs.base.stat ssh: false, sudo: false, target: config.target
+            return stats if utils.stats.isFile stats.mode
+            throw Error "Invalid target: #{config.target}"
+          catch err
+            return null if err.code is 'NIKITA_FS_STAT_TARGET_ENOENT'
+            throw err
+        catch err
+          return null if err.code is 'NIKITA_FS_STAT_TARGET_ENOENT'
+          throw err
+      # Now that we know the real name of the target, define a temporary file to write
+      stage_target = "#{config.target}.#{Date.now()}#{Math.round(Math.random()*1000)}"
+      {status} = await @call ->
+        return true unless stats
+        {hash} = await @fs.hash target: config.source, algo: algo
+        hash_source = hash
+        {hash} = await @fs.hash target: config.target, algo: algo, ssh: false, sudo: false
+        hash_target = hash
+        match = hash_source is hash_target
+        log if match
+        then message: "Hash matches as '#{hash_source}'", level: 'INFO', module: 'nikita/lib/file/download' 
+        else message: "Hash dont match, source is '#{hash_source}' and target is '#{hash_target}'", level: 'WARN', module: 'nikita/lib/file/download'
+        not match
+      return unless status
+      @fs.mkdir
+        ssh: false
+        sudo: false
+        target: path.dirname stage_target
+      @fs.base.createReadStream
+        target: config.source
+        stream: (rs) ->
+          ws = fs.createWriteStream stage_target
+          rs.pipe ws
+      @fs.move
+        ssh: false
+        sudo: false
+        source: stage_target
+        target: config.target
+      log message: "Unstaged uploaded file", level: 'INFO', module: 'nikita/lib/file/upload'
```

Comment:
in this case, log is called before the stream was consume and the file written, need `await` (see the `fs.base.createReadStream` tests